opensearch_py_ml/ml_commons/ml_commons_client.py

```diff
@@ -509,7 +509,7 @@ def generate_sparse_encoding(self, model_id: str, sentences: List[str]) -> objec
         :return: Returns a JSON object `inference_results` containing the sparse encoding results for the given sentences.
         :rtype: object
         """
-        API_URL = f"{ML_BASE_URI}/_predict/sparse_encoding/{model_id}"
+        API_URL = f"{ML_BASE_URI}/models/{model_id}/_predict/"
```

Comment:
What I was trying to mean if `_predict` method works for all models, then we can have a generic `predict` method here. We don't need `generate_embedding` or `generate_sparse_encoding` method, right?

Reply:
agreed. If we are using request_body as dict, then it can also be used to sentence transformers.